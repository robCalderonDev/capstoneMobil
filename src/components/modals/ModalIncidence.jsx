import React, { useState } from 'react';
import { Alert, Modal, StyleSheet, Text, Pressable, View, Image } from 'react-native';
import { TouchableOpacity } from 'react-native';

const ModalIncidence = ({ modalVisible, setModalVisible, itemSelected }) => {

    return (
        <View style={styles.centeredView}>
            <Modal
                animationType="slide"
                transparent={true}
                visible={modalVisible}
                onRequestClose={() => {

                    setModalVisible(!modalVisible);
                }}>
                <View style={styles.centeredView}>
                    <View style={styles.modalView}>
                        <View style={styles.titleContainer}>
                            <Text style={styles.modalTitle}>{itemSelected.titulo}</Text>
                            {/* <Image
                                source={require('../../img/check.png')}
                                style={{
                                    width: 25, // Ajusta el ancho de la imagen
                                    height: 25, // Ajusta la altura de la imagen
                                    marginTop: -2.5,
                                    marginLeft: 10,

                                }}
                            /> */}
                        </View>

                        <Text style={styles.modalText}>{itemSelected.descripcion}<Text style={styles.mail}></Text> curso: 2B </Text>
                        <View style={styles.buttonContainer}>
                            <TouchableOpacity style={styles.touchableOpacity1} onPress={() => {

                                setModalVisible(!modalVisible);

                            }}>
                                <Text style={styles.text}>Vale</Text>
                            </TouchableOpacity>

                        </View>

                    </View>
                </View>
            </Modal>

        </View>
    );
};

const styles = StyleSheet.create({

    buttonContainer: {
        flexDirection: 'row',
        justifyContent: 'space-around',
    },
    text: {
        color: 'white',
        fontSize: 15,
        textAlign: 'center',
        paddingTop: 8,
    },
    centeredView: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 22,
    },
    modalView: {
        margin: 20,
        backgroundColor: '#313338',
        borderRadius: 15,
        padding: 35,
        width: '80%',
        minHeight: '30%',
        alignItems: 'center',



    },
    touchableOpacity1: {
        backgroundColor: '#56AF84',

        height: 40,
        width: '40%',
        borderRadius: 2,

    },
    modalTitle: {
        marginBottom: 15,
        textAlign: 'center',
        fontSize: 17,
        color: 'white',
        fontWeight: 'bold',
    },
    modalText: {
        marginBottom: 15,
        textAlign: 'center',
        fontSize: 15,
        color: 'white',

    },
    titleContainer: {
        flexDirection: 'row',
        justifyContent: 'space-around',
    }

});

export default ModalIncidence;