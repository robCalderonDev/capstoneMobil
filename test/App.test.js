import renderer from 'react-test-renderer'
import ModalIncidence from '../src/components/modals/ModalIncidence';

import React, { useState } from 'react'
describe('<App />', () => {
    it('has 1 child', () => {
        const tree = renderer.create(<ModalIncidence />).toJSON();
        expect(tree.children.length).toBe(1);
    });
    it("renders correctly", () => {
        const tree = renderer.create(<ModalIncidence />).toJSON();
        expect(tree).toMatchSnapshot()
    })
});
