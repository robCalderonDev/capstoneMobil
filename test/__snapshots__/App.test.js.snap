// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#313338",
            "borderRadius": 15,
            "margin": 20,
            "minHeight": "30%",
            "padding": 35,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-around",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 17,
                "fontWeight": "bold",
                "marginBottom": 15,
                "textAlign": "center",
              }
            }
          />
        </View>
        <View />
        <Text
          style={
            {
              "color": "white",
              "fontSize": 15,
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
        >
            
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-around",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#56AF84",
                "borderRadius": 2,
                "height": 40,
                "opacity": 1,
                "width": "40%",
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 15,
                  "paddingTop": 8,
                  "textAlign": "center",
                }
              }
            >
              Vale
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
